### PR TITLE
Add `Backup` resource custom hooks, terminalCodes and e2e tests

### PR DESCRIPTION
**Pull request by A-Hilaly**
Part of https://github.com/aws-controllers-k8s/community/issues/803

Description of changes:
- Add custom hooks, terminalCodes to `Backup` resource in `generator.yaml`
- Add e2e tests for `Backup` create and delete operations

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by A-Hilaly** [comment]
Note to reviewers: This PR needs to be rebased once #3 and #4 are merged
/hold

File: pkg/resource/backup/hooks.go
Comment (jaypipes) [inline pkg/resource/backup/hooks.go:27]: this is meant to be empty?

Comment (A-Hilaly) [reply pkg/resource/backup/hooks.go:27]: There are no terminal statuses for `Backup`, for now.. https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_BackupDetails.html

File: test/e2e/tests/test_backup.py
Comment (RedbackThomson) [inline test/e2e/tests/test_backup.py:96]: Oops

**Review by RedbackThomson — APPROVED** [review]

File: pkg/resource/backup/hooks.go
Comment (jaypipes) [reply pkg/resource/backup/hooks.go:27]: k.

File: test/e2e/tests/test_backup.py
Comment (A-Hilaly) [reply test/e2e/tests/test_backup.py]: Nice catch!

**Comment by A-Hilaly** [comment]
Mmmmm it looks like tests are randomly failing from time to time.. investigating that.
/hold

**Review by jaypipes — APPROVED** [review]
tres bien, merci!

**Pull request merged**
